backend/src/plugins/Automod/actions/setSlowmode.ts

```diff
@@ -1,33 +1,32 @@
-import { Snowflake, TextChannel } from "discord.js";
+import { Snowflake, TextChannel, ThreadChannel } from "discord.js";
 import * as t from "io-ts";
 import { ChannelTypeStrings } from "src/types";
-import { LogType } from "../../../data/LogType";
 import { convertDelayStringToMS, isDiscordAPIError, tDelayString, tNullable } from "../../../utils";
 import { automodAction } from "../helpers";
 import { LogsPlugin } from "../../Logs/LogsPlugin";
 
 export const SetSlowmodeAction = automodAction({
   configType: t.type({
-    channels: t.array(t.string),
+    channels: tNullable(t.array(t.string)),
     duration: tNullable(tDelayString),
   }),
 
   defaultConfig: {
     duration: "10s",
   },
 
-  async apply({ pluginData, actionConfig }) {
+  async apply({ pluginData, actionConfig, contexts }) {
     const slowmodeMs = Math.max(actionConfig.duration ? convertDelayStringToMS(actionConfig.duration)! : 0, 0);
-
-    for (const channelId of actionConfig.channels) {
+    const channels: string[] = actionConfig.channels ?? [];
+    if (channels.length === 0) {
+      channels.push(...contexts.filter(c => c.message?.channel_id).map(c => c.message!.channel_id));
+    }
+    for (const channelId of channels) {
       const channel = pluginData.guild.channels.cache.get(channelId as Snowflake);
-
       // Only text channels and text channels within categories support slowmodes
-      if (!channel || !(channel.type === ChannelTypeStrings.TEXT || ChannelTypeStrings.CATEGORY)) {
-        continue;
-      }
+      if (!channel?.isText() && channel?.type !== ChannelTypeStrings.CATEGORY) continue;
 
-      const channelsToSlowmode: TextChannel[] = [];
+      const channelsToSlowmode: Array<TextChannel | ThreadChannel> = [];
```

Comment:
For preserving consistency with the rest of the code, do you think it'd be better to use `(TextChannel | ThreadChannel)[]`?

Reply:
TSLint disallows it